spec/lib/tasks/provider_transfer/ptr_migrations_spec.rb

```diff
@@ -0,0 +1,209 @@
+require "rails_helper"
+
+RSpec.describe "ptr_migrations:", type: :task do
+  before do
+    Rails.application.load_tasks if Rake::Task.tasks.empty?
+  end
+
+  describe "ptr_migrations:laa_transfer_office" do
+    subject(:task) do
+      Rake::Task["ptr_migrations:laa_transfer_office"]
+    end
+
+    let(:initial_office) { create(:office) }
+    let(:new_office) { create(:office) }
+    let(:legal_aid_application) { create(:legal_aid_application, office: initial_office) }
+
+    context "when LegalAidApplication and Office exist" do
+      it "transfers the LegalAidApplication to the new office" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) current office code: #{initial_office.code}")
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) successfully transferred to office code: #{new_office.code}")
+
+        task.execute(mock: "false", app_id: legal_aid_application.id, office_id: new_office.id)
+
+        expect(legal_aid_application.reload.office).to eq(new_office)
+      end
+    end
+
+    context "when LegalAidApplication does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exist")
+
+        task.execute(mock: "false", app_id: nil, office_id: new_office.id)
+
+        expect(legal_aid_application).not_to receive(:update!)
+        expect(legal_aid_application.reload.office).to eq(initial_office)
+      end
+    end
+
+    context "when Office does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exist")
+
+        task.execute(mock: "false", app_id: legal_aid_application.id, office_id: nil)
+
+        expect(legal_aid_application).not_to receive(:update!)
+        expect(legal_aid_application.reload.office).to eq(initial_office)
+      end
+    end
+
+    context "when both LegalAidApplication and Office do not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exist")
+
+        task.execute(mock: "false", app_id: nil, office_id: nil)
+
+        expect(legal_aid_application).not_to receive(:update!)
```

Comment:
[ditto](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/7591/files#r1942985796)